nerfstudio/viewer/viewer.py

```diff
@@ -282,8 +289,149 @@ def nested_folder_install(folder_labels: List[str], prev_labels: List[str], elem
+        min_corner = min_coords
+        max_corner = max_coords
+
+        vertices = np.array(
+            [
+                [min_corner[0], min_corner[1], min_corner[2]],  # Vertex 0
+                [max_corner[0], min_corner[1], min_corner[2]],  # Vertex 1
+                [max_corner[0], max_corner[1], min_corner[2]],  # Vertex 2
+                [min_corner[0], max_corner[1], min_corner[2]],  # Vertex 3
+                [min_corner[0], min_corner[1], max_corner[2]],  # Vertex 4
+                [max_corner[0], min_corner[1], max_corner[2]],  # Vertex 5
+                [max_corner[0], max_corner[1], max_corner[2]],  # Vertex 6
+                [min_corner[0], max_corner[1], max_corner[2]],  # Vertex 7
+            ]
+        )
+
+        faces = np.array(
+            [
+                [0, 1, 2],
+                [0, 2, 3],  # Bottom face
+                [4, 5, 6],
+                [4, 6, 7],  # Top face
+                [0, 1, 5],
+                [0, 5, 4],  # Side face
+                [1, 2, 6],
+                [1, 6, 5],  # Side face
+                [2, 3, 7],
+                [2, 7, 6],  # Side face
+                [3, 0, 4],
+                [3, 4, 7],  # Side face
+            ]
+        )
+
+        return vertices, faces
+
+    def set_callback_in_closure(self, i, control):
+        @control.on_update
+        def _(_):
+            self.update_bounding_box()
+
+    def update_bounding_box(self):
+        min_coords = np.minimum(self.transform_controls[0].position, self.transform_controls[1].position)
+        max_coords = np.maximum(self.transform_controls[0].position, self.transform_controls[1].position)
+        vertices, faces = Viewer.convert_aabb_to_bounding_box(min_coords, max_coords)
+
+        # Add or update the bounding box mesh in the Viser scene
+        self.viser_server.add_mesh_simple(
+            name="/bounding_box",
+            vertices=vertices,
+            faces=faces,
+            color=(255, 255, 255),
+            wireframe=True,
+            opacity=1.0,
+            material="standard",
+            flat_shading=False,
+            side="double",
+        )
+
+        # Update the GUI elements with the new positions
+        for i, control in enumerate(self.transform_controls):
+            self.gui_point_positions[i].value = control.position
+
+    def update_scene_box(self):
+        min_coords = np.minimum(self.transform_controls[0].position, self.transform_controls[1].position)
+        max_coords = np.maximum(self.transform_controls[0].position, self.transform_controls[1].position)
+
+        new_aabb = torch.tensor(
+            [min_coords / VISER_NERFSTUDIO_SCALE_RATIO, max_coords / VISER_NERFSTUDIO_SCALE_RATIO],
+            device=self.pipeline.model.device,
+            requires_grad=False,
+        )
+
+        self.pipeline.model.scene_box = SceneBox(aabb=new_aabb)
+        self.pipeline.model.render_aabb = SceneBox(aabb=new_aabb)
+        self.pipeline.model.populate_modules()
```

Comment:
Using `device=self.pipeline.model.device` for `new_aabb` seems to cause issues where tensors end up on different devices later in the process. I'm not entirely sure what the best solution is to address this.

@brentyi @tancik, since you both work with this part of the codebase frequently, do you have any suggestions?